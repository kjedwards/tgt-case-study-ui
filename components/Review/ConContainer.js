/* ----------------------------------------------
 * Component to display the header for cons
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  width: 50%;
  float: right;
`

const Header = styled.div`
  font-size: 24px;
`

const Content = styled.div`
  font-size: 11px;
`


class ConContainer extends Component {
    render () {
      return(
        <Container>
          <Header>CON</Header>
          <Content>most helpful 1-2 star review</Content>
        </Container>
      )
    }
}


export default ConContainer
