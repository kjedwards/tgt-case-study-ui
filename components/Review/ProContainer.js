/* ----------------------------------------------
 * Component to display the header for pros
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  width: 50%;
  float: left;
`

const Header = styled.div`
  font-size: 24px;
`

const Content = styled.div`
  font-size: 11px;
`


class ProContainer extends Component {
    render () {
      return(
        <Container>
          <Header>Pro</Header>
          <Content>most helpful 4-5 star review</Content>
        </Container>
      )
    }
}


export default ProContainer
