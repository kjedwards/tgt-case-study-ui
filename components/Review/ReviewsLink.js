/* ----------------------------------------------
 * Component to display tthe link to the review page
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/
