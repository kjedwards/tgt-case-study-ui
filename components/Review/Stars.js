/* ----------------------------------------------
 * Component to display the stars for a review
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'
import 'font-awesome/css/font-awesome.min.css'

const Container = styled.div`
  display: flex;
`

const Unchecked = styled.div`
  color: grey;
`

const Checked = styled.div`
  color: red;
`

class Stars extends Component {

  constructor(props) {
    super(props)
  }

    render () {
      //Set up array of checked stars
      var checked = [], i = 0, len = this.props.rating;
      while (++i <= len) checked.push(i);

      //Set up array of unchecked stars
      var unchecked = [], i = 0, len = 5 - this.props.rating;
      while (++i <= len) unchecked.push(i);

      return(
        <Container>
          <Checked>
            {checked.map((x, i) =>
              <i className='fa fa-star' key={i} />
            )}
          </Checked>
          <Unchecked>
            {unchecked.map((x, i) =>
              <i className='fa fa-star' key={i} />
            )}
          </Unchecked>
        </Container>
      )
    }
}


export default Stars
