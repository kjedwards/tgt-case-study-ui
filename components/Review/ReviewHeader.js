/* ----------------------------------------------
 * Component to contain pro and con header
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import ProContainer from './ProContainer'
import ConContainer from './ConContainer'

const RvwHeader = styled.div`
  margin: 10px;
  height: 60px;
  border-bottom: 1px solid lightgrey;
`


class ReviewHeader extends Component {
    render () {
      return(
        <RvwHeader>
          <ProContainer></ProContainer>
          <ConContainer></ConContainer>
        </RvwHeader>
      )
    }
}


export default ReviewHeader
