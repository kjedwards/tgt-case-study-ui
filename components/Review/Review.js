/* ----------------------------------------------
 * Component to display the actual review
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import Stars from './Stars'

const Container = styled.div`
  margin: 10px;
`

const StarContainer = styled.div`
  margin: 5px;
  font-size: 1em;
`

const ReviewTitle = styled.div`
  margin: 5px;
  font-weight: bold;
  font-size: 16px;
`

const ReviewBody = styled.div`
  font-size: 12px;
  margin: 5px;
  margin-left: 8px;
`

const ReviewFooter = styled.div`
  font-size: 10px
  margin: 5px;
  margin-left: 8px;
`

const Name = styled.a`
  color: #6994c1;
  margin-top: 5px;
  margin-right: 5px;
`

const Date = styled.div`

`

class Review extends Component {
    render () {

      return(
        <Container>
          <StarContainer>
            <Stars rating={this.props.review.data.overallRating}></Stars>
          </StarContainer>
          <ReviewTitle>
            {this.props.review.data.title}
          </ReviewTitle>
          <ReviewBody>
            {this.props.review.data.review}
          </ReviewBody>
          <ReviewFooter>
            <Name>{this.props.review.data.screenName}</Name>
            <Date>{this.props.review.data.datePosted}</Date>
          </ReviewFooter>
        </Container>
      )
    }
}


export default Review
