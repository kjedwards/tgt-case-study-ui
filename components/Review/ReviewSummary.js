/* ----------------------------------------------
 * Component to contain all but the overall stars
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import ReviewHeader from './ReviewHeader'
import Review from './Review'

const RvwSummary = styled.div`
  background-color: #f7f7f7;
  width:100%;
`

const RvwHolder = styled.div`
  display: flex;
`

const ProRvw = styled.div`
  width: 45%
  margin: 5px;
  margin-left: 0px;
`

const ConRvw = styled.div`
  width: 50%;
  margin: 5px;
  margin-left: 0px;
`

class ReviewSummary extends Component {
    render () {
      return(
        <RvwSummary>
          <ReviewHeader></ReviewHeader>
          <RvwHolder>
            <ProRvw>
              <Review review={this.props.pro}></Review>
            </ProRvw>
            <ConRvw>
              <Review review={this.props.con}></Review>
            </ConRvw>
          </RvwHolder>
        </RvwSummary>
      )
    }
}


export default ReviewSummary
