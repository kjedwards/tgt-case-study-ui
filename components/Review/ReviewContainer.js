import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../../constants/DeviceWidth'

import ReviewTop from './ReviewTop'
import ReviewSummary from './ReviewSummary'

const RvwContainer = styled.div`
  background-color: green;
  width: 40%;
  min-width: 400px;
  order: 1;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;

  @media(max-width: ${sizes.tablet}) {
    order: 2;
  }
`


class ReviewContainer extends Component {
    render () {
      return(
        <RvwContainer>
          <ReviewTop></ReviewTop>
          <ReviewSummary></ReviewSummary>
        </RvwContainer>
      )
    }
}


export default ReviewContainer
