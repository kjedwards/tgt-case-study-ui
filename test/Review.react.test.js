import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

//Imported Components
import ConContainer from '../components/Review/ConContainer'
import ProContainer from '../components/Review/ProContainer'
import Review from '../components/Review/Review'
import ReviewHeader from '../components/Review/ReviewHeader'
import ReviewSummary from '../components/Review/ReviewSummary'
import Stars from '../components/Review/Stars'

//Mocks
jest.mock('../components/Review/ConContainer', () => 'con-container')
jest.mock('../components/Review/ProContainer', () => 'pro-container')
jest.mock('../components/Review/Review', () => 'review')
jest.mock('../components/Review/ReviewHeader', () => 'review-header')
jest.mock('../components/Review/Stars', () => 'stars')

describe('ConContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ConContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ConContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ProContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ProContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ProContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Review Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Review />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Review />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Review Header Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ReviewHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ReviewHeader />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Review Summary Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Review Summary />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Review Summary />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Stars Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Stars />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Stars />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
