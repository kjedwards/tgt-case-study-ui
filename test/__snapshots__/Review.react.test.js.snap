// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConContainer Component ENZYME snapshot 1`] = `<con-container />`;

exports[`ConContainer Component RENDERER snapshot 1`] = `<con-container />`;

exports[`ProContainer Component ENZYME snapshot 1`] = `<pro-container />`;

exports[`ProContainer Component RENDERER snapshot 1`] = `<pro-container />`;

exports[`Review Component ENZYME snapshot 1`] = `<review />`;

exports[`Review Component RENDERER snapshot 1`] = `<review />`;

exports[`Review Header Component ENZYME snapshot 1`] = `<review-header />`;

exports[`Review Header Component RENDERER snapshot 1`] = `<review-header />`;

exports[`Review Summary Component ENZYME snapshot 1`] = `
<review
  Summary={true}
/>
`;

exports[`Review Summary Component RENDERER snapshot 1`] = `
<review
  Summary={true}
/>
`;

exports[`Stars Component ENZYME snapshot 1`] = `<stars />`;

exports[`Stars Component RENDERER snapshot 1`] = `<stars />`;
